.github/workflows/npm-audit.yml.dist

```diff
@@ -0,0 +1,82 @@
+name: Npm audit
+
+on:
+  schedule:
+    - cron: '0 12 * * 0'  # Run every fortnight on Sunday at 12
+
+jobs:
+  npm_audit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Use Node.js from .nvmrc in modules/custom
+        id: npm_audit_modules
+        run: |
+          find public/modules/custom -type f -name ".nvmrc" -exec sh -c '
+            dir=$(dirname "$1")
+            node_version=$(cat "$1")
+            echo "Using Node.js version $node_version in $dir"
+            cd "$dir"
+            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+            export NVM_DIR="$HOME/.nvm"
+            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+            nvm install $node_version
+            nvm use $node_version
+            npm install --silent
+            set +e
+            npm audit --package-lock-only --loglevel=error;
+            # The npm audit command will exit with a 0 exit code if no vulnerabilities were found.
+            if [ $? -gt 0 ]; then npm audit fix --package-lock-only --loglevel=error; echo "CREATE_PR=true" >> $GITHUB_OUTPUT; fi;
+            set -e
+          ' sh {} \;
+
+      - name: Use Node.js from .nvmrc in themes/custom
+        id: npm_audit_themes
+        run: |
+          find public/themes/custom -type f -name ".nvmrc" -exec sh -c '
+            dir=$(dirname "$1")
+            node_version=$(cat "$1")
+            echo "Using Node.js version $node_version in $dir"
+            cd "$dir"
+            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+            export NVM_DIR="$HOME/.nvm"
+            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+            nvm install $node_version
```

Comment:
Fyi, `nvm` should already be installed, so this whole nvm setup can be replaced with `export NVM_DIR="$HOME/.nvm" && [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"`